python/doc/theory/meta_modeling/regression_analysis.rst

```diff
@@ -0,0 +1,197 @@
+  basis functions,
+- :math:`\{a_k \in \Rset\}_{k = 1, ..., p}` are the coefficients,
+- :math:`\epsilon \sim \cN(0, \sigma^2)` where
+  :math:`\cN` is a normal distribution and
+  :math:`\sigma > 0` is its standard deviation.
+
+Caution
+~~~~~~~
+
+There is an ambiguity when the number of parameters is unspecified in the
+text, which may explain the differences between the various formulas
+we find in the books.
+In some texts, the intercept has the index 0, which leads to an
+increased number of parameters.
+In the present document, the number of parameters is equal to :math:`p`,
+but not all books use the same convention:
+
+- in [baron2014]_ (eq. 11.11 page 396), in [sen1990]_ (eq. 2.1 page 28)
+  and in [rawlings2001]_ (eq. 3.1 page 75),
+  the number of parameters is equal to :math:`p + 1`,
+- in [bingham2010]_ (page 61) and in [faraway2014]_ (page 15),
+  the number of parameters is equal to :math:`p`.
+
+Experimental design
+~~~~~~~~~~~~~~~~~~~
+
+A set of observations of the input parameters is required:
+
+.. math::
+
+    \cX = \left\{ \vect{x}^{(1)}, \dots, \vect{x}^{(N)} \right\},
+
+as well as the corresponding model evaluations:
+
+.. math::
+
+    \cY = \left\{ y^{(1)},\dots,y^{(N)} \right\}.
+
+Let :math:`\vect{y} = (y^{(1)},\dots,y^{(N)})^T \in \Rset^{N}` be the
+vector of output observations.
+
+Statistics
+~~~~~~~~~~
+
+Let :math:`\bar{y}` be the sample mean:
+
+.. math::
+
+    \bar{y} = \frac{1}{n} \sum_{j = 1}^n y_j.
+
+The total sum of squares ([baron2014]_ page 398) is:
+
+.. math::
+
+    SS_{TOT}
+    = \sum_{j = 1}^n \left(y_j - \bar{y}_j\right)^2
+    = \left(\vect{y} - \bar{\vect{y}}\right)^T \left(\vect{y} - \bar{\vect{y}}\right).
+
+where :math:`\hat{\vect{y}} = (\bar{y}, ..., \bar{y})^T \in \Rset^n`.
+
+The regression sum of squares is:
+
+.. math::
+
+    SS_{REG}
+    = \sum_{j = 1}^n \left(\hat{y}_j - \bar{y}_j\right)^2
+    = \left(\hat{\vect{y}} - \bar{\vect{y}}\right)^T \left(\hat{\vect{y}} - \bar{\vect{y}}\right).
+
+The error sum of squares is:
+
+.. math::
+
+    SS_{ERR}
+    = \sum_{j = 1}^n \left(y_j - \hat{y}_j\right)^2
+    = \left(\vect{y} - \hat{\vect{y}}\right)^T \left(\vect{y} - \hat{\vect{y}}\right).
+
+Coefficient of determination
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The coefficient of determination is ([baron2014]_ page 399):
+
+.. math::
+
+    R^2 = \frac{SS_{REG}}{SS_{TOT}}.
+
+The coefficient of determination measures the part
+of the variance explained by the linear regression model.
+
+Variance
+~~~~~~~~
+
+The unbiased estimator of the variance is ([baron2014]_ page 400,
+[bingham2010]_ page 67):
+
+.. math::
+
+    \hat{\sigma}^2 = \frac{SS_{ERR}}{n - p}.
+
+ANOVA F-test
+~~~~~~~~~~~~
+
+The F-statistic is based on the hypothesis that all coefficients
+are simultaneously zero ([baron2014]_ page 400).
+The ANOVA F-test is based on the hypothesis:
+
+.. math::
+
+    H_0 : a_1 = . . . = a_p = 0
+    \qquad \textrm{vs} \qquad
+    H_A : \textrm{at least one } a_k \neq 0.
```

Comment:
Aren't you basically writing the same thing twice?

Reply:
I don't think so. H0 is the hypothesis that all coefficients are zero. HA is the hypothesis that at least one coefficient is nonzero. Does it make sense?

Reply:
You are right: the text is confusing. I meant a first sentence to introduce the test using words, then a second sentence using mathematics. Fixed at 4b53aeb3de2f4269da1a1fb73b79350b17636541.